Feat/theme switcher 17

khushthecoder opened the pull request:
## Description
This PR activates and refines the light theme, making the existing dark/light theme toggle fully functional. It ensures all components are styled correctly and provides a polished, consistent user experience across both themes, as requested in the issue.

## Related Issue
Fixes #17

## Changes Made
- **Refined Light Theme Colors:** Updated the CSS variables in `src/app/globals.css` under the `:root` selector to provide a more modern and visually appealing light theme.
- **Improved Readability:** Adjusted background, foreground, and primary colors for better contrast and a softer, more professional look, moving away from pure white.
- **Ensured Theming Consistency:** Verified that the existing theme toggle in the `Header` component now correctly switches between the polished dark theme and the newly refined light theme.

## Screenshots or GIFs (if applicable)

**New Light Theme:**
<img width="1512" height="862" alt="image" src="https://github.com/user-attachments/assets/373f25f9-5111-4a88-9c20-8575d8fb1cc7" />

**Dark Theme (For Comparison):**
<img width="1512" height="862" alt="image" src="https://github.com/user-attachments/assets/69c2cd8d-abeb-4970-9233-3147e522cd91" />
## Checklist

- [x] Only the necessary files are modified; no unrelated changes are included.
- [x] Follows clean code principles (readable, maintainable, minimal duplication).
- [x] All changes are clearly documented.
- [x] Code has been tested (manual/automated) and verified against edge cases.
- [x] No breaking changes are introduced to existing functionality.
- [ ] All new and existing tests passed (if tests exist).

## Additional Notes
The core infrastructure for theme switching (`next-themes` and the UI toggle) was already present. This PR focuses on refining the light theme's color palette to make the feature production-ready.

All components have been manually tested in both light and dark modes to ensure visual consistency.

fahimahammed commented:
:wave: Hi everyone,

This discussion space is dedicated to **DevUI**! 
We’d love to hear from you and get the community involved.

You can share your thoughts in different categories:

* :bulb: **Ideas** – Suggest new components, features, or improvements
* :person_raising_hand: **Q&A** – Ask questions and get help from the community
* :ballot_box: **Polls** – Vote and share your opinions on upcoming decisions
* :loudspeaker: **General** – Open conversations, announcements, and feedback

Jump in and start a discussion here :point_right: https://github.com/fahimahammed/DevUI/discussions

Your feedback will help shape the future of **DevUI**

khushthecoder commented:
Hi  @fahimahammed ,
Following up on my previous message, I've now updated the Pull Request with the improvements I mentioned.
To make the theming feature more substantial, I've implemented a live, persistent theme color picker.

This new feature includes:
Real-Time Customization: A new paintbrush icon allows users to instantly change the site's primary accent color from a curated palette.
Persistence: The user's color choice is automatically saved to their browser's local storage, ensuring their preferred theme loads on every visit.

I believe these additions make the theming functionality complete, dynamic, and user-friendly.
Thanks for the guidance. The PR is now ready for another review when you have a moment.

khushthecoder commented:
Hi @fahimahammed ,

I wanted to let you know that I’ve already resolved the merge conflict once, but after the latest updates from main, two more conflicts appeared which I have also resolved. At this point, the branch is fully updated and ready, but unless the PR is reviewed and either merged or closed, new conflicts will likely continue to arise as main changes.

Could you please take a look at the PR when you get a chance? Thanks!

khushthecoder commented:
Hi @fahimahammed ,
I've updated the branch and resolved all merge conflicts.
The PR is now up-to-date and ready for your review. Thanks!
